crates/bevy_asset/src/path.rs

```diff
@@ -63,6 +63,51 @@ impl<'a> AssetPath<'a> {
                 .map(|value| Cow::Owned(value.to_string())),
         }
     }
+
+    /// Resolves a relative asset path. The result will be an `AssetPath` which is resolved relative
+    /// to this path. There are three cases:
+    ///
+    /// If the relative path begins with `#`, then it is considered an asset label, in which case
+    /// the result is the base path with the label portion replaced.
+    ///
+    /// If the relative path starts with "./" or "../", then the result is combined with the base
+    /// path. The rules are the same as for URIs, which means that the relative path is resolved
+    /// relative to the *directory* of the base path, so a base path of `"x/y/z#foo"` combined with
+    /// a relative path of `"./a#bar"` yields `"x/y/a#bar"`.
+    ///
+    /// If neither of the above are true, then the relative path is considered a 'full' path,
+    /// and the result is simply a copy of the relative path, converted into an `AssetPath`.
+    /// Note that a 'full' asset path is still relative to the asset root directory, and not
+    /// necessarily an absolute filesystem path.
+    pub fn resolve(&'a self, relative_path: &'a str) -> AssetPath<'a> {
+        if let Some(without_prefix) = relative_path.strip_prefix('#') {
+            AssetPath::new_ref(&self.path, Some(without_prefix))
+        } else if relative_path.starts_with("./") || relative_path.starts_with("../") {
```

Comment:
I think another way of doing this could be extracting out any label and then converting the rest of the path to a `Path` reference. Then you can use methods like `join`, `parent`, and `components` to resolve the relative path.

Not sure if that would be faster/easier, but I think it could probably be cleaner than a bunch of string operations.